packages/core/lib/segments/attributes/subsegment.js

```diff
@@ -122,19 +122,19 @@ Subsegment.prototype.addPrecursorId = function(id) {
  */
 
 Subsegment.prototype.addAnnotation = function(key, value) {
-  if (!(typeof value === 'boolean' || typeof value === 'string' || (typeof value === 'number' && isFinite(value)))) {
-    throw new Error('Failed to add annotation key: ' + key + ' value: ' + value + ' to subsegment ' +
+  if (typeof value !== 'boolean' && typeof value !== 'string' && !isFinite(value)) {
+    logger.getLogger().error('Failed to add annotation key: ' + key + ' value: ' + value + ' to subsegment ' +
       this.name + '. Value must be of type string, number or boolean.');
```

Comment:
My thought was it would be easier for diagnosising the problem if we still added this data but indicated the problem via how it was stored, since if configuration isn't set to log we won't know why it's missing.

For instance when you see:
```json
[Object object]: data
```

You know you messed up the key type,

If you see:
```json
"key": ''
```

That tells you something about the data. We could also store.

```json
"key": "X-Ray::InvalidValue"
```

That would be even better.

Reply:
I would still like to take my suggested path, but in the interested of avoiding scope creep here. I've converted these to `return` instead.

Depending on your guidance, we can decide whether or not to also make my suggestion here have a followup.

Reply:
> I would still like to take my suggested path, but in the interested of avoiding scope creep here. I've converted these to return instead.

This sounds good. Given that we have a limit of 50 annotations per trace, I don't think we want to add these InvalidValue errors on customer's behalf & eat into their limit.

Reply:
>  I don't think we want to add these InvalidValue errors on customer's behalf & eat into their limit.

Okay, but this doesn't really matter, because they intended to use one up on this key + valid value, so while we don't have to make the change here, that's not a good argument against it.